SRC/TableExtensions/OVWExcelBuffer.TableExtension.al

```diff
@@ -0,0 +1,164 @@
+tableextension 50131 "OVW Excel Buffer" extends "Excel Buffer"
+{
+    fields
+    {
+        field(50131; "OVW Value is BLOB"; Boolean)
+        {
+            DataClassification = SystemMetadata;
+        }
+    }
+
+    procedure AddCell(pintRow: Integer; var pintColumn: Integer; pvarValue: Variant; ptxtHeading: Text[250]; pintColumnIncrement: Integer)
+    begin
+        AddCellWithFormatting(pintRow, pintColumn, pvarValue, ptxtHeading, false, false, false, 2, pintColumnIncrement); //34764+-
+    end;
+
+    procedure AddCellWithFormatting(RowNo: Integer; var ColumnNo: Integer; CellValue: Variant; CellHeading: Text[250]; IsBold: Boolean; IsItalic: Boolean; IsUnderlined: Boolean; NumberOfDecimals: Integer; ColumnIncrementValue: Integer)
+    var
+        DecimalNotation: Text;
+    begin
+        Init();
+        Validate("Row No.", RowNo);
+        Validate("Column No.", ColumnNo);
+        TestField("Row No.");
+        TestField("Column No.");
+        Formula := '';
+        Bold := IsBold;
+        Italic := IsItalic;
+        Underline := IsUnderlined;
+
+        DecimalNotation := '#,##0';
+        if NumberOfDecimals > 0 then begin
+            DecimalNotation += '.';
+            DecimalNotation := PadStr(DecimalNotation, StrLen(DecimalNotation) + NumberOfDecimals, '0');
+        end;
+
+        case true of
+            CellValue.IsInteger,
+            CellValue.IsDecimal,
+            CellValue.IsDuration,
+            CellValue.IsBigInteger:
+                begin
+                    NumberFormat := '0';
+                    "Cell Type" := "Cell Type"::Number;
+                    if CellValue.IsDecimal then
+                        NumberFormat := DecimalNotation;
+                end;
+
+            CellValue.IsDate:
+                begin
+                    "Cell Type" := "Cell Type"::Date;
+                end;
+
+            CellValue.IsTime:
+                begin
+                    "Cell Type" := "Cell Type"::Time;
+                end;
+
+            CellValue.IsDateTime:
+                begin
+                    "Cell Type" := "Cell Type"::Date;
+                end;
+
+            else begin
+                "Cell Type" := "Cell Type"::Text;
+                if StrLen(Format(CellValue)) > MaxStrLen("Cell Value as Text") then begin
+                    "OVW Value is BLOB" := true;
+                    ConvertStringToBlob(Format(CellValue), TempBlob);
+                    TempBlob.FromRecord(Rec, FieldNo("Cell Value as Blob"));
+                end;
+            end;
+        end;
+        if not "OVW Value is BLOB" then
+            "Cell Value as Text" := Format(CellValue);
+        if not Insert() then
+            Modify();
+        AddHeading(ColumnNo, CellHeading);
+        ColumnNo += ColumnIncrementValue;
+    end;
+
+    [Obsolete('This procedure is no longer used and will be deprecated. This warning will become an error in the future.', '19.1.20220408')]
```

Comment:
Should this Obsolete tag be here? If so, remove the procedure and if not, please remove the tag.

And again, please refactor variable names to adhere to the Microsoft coding guidelines. For details please check the [AL Guidelines page.](https://alguidelines.dev/)